tests/test_sandbox/test_ecs_fargate.py

```
@@ -757,3 +849,282 @@ def test_ensure_image_built_cache_hit(self, mock_exists, tmp_path):
+            return {"taskDefinition": {"taskDefinitionArn": "arn:aws:ecs:us-east-1:1234:task-definition/test:1"}}
+
+        sb._ecs.register_task_definition.side_effect = slow_register
+
+        payload = {"family": "test", "cpu": "4096", "memory": "8192", "containerDefinitions": []}
+        results: list[str] = []
+        errors: list[BaseException] = []
+
+        def call():
+            try:
+                results.append(sb._do_register(payload))
+            except BaseException as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `BaseException`

(BLE001)